### PR TITLE
Fix MapboxOverlay scale issue when device pixel ratio changes

### PR DESCRIPTION
**Pull request by Pessimistress**
For #7857

Mapbox, Maplibre and deck.gl each treat device pixel ratio change differently. Let the base map do the scaling when using interleaved mode.

#### Change List
- `Deck` class: disable `autoResizeDrawingBuffer` if using an external gl context (AFAIK this is only used when interleaving with Mapbox/Google Maps/ArcGIS)
- Bump luma dependency for https://github.com/visgl/luma.gl/pull/1746

### CODE REVIEW

**Comment by chrisgervang** [comment]
Does this change the way to manually control the resolution in interleaved mode?

**Comment by Pessimistress** [comment]
You shouldn't be able to control the resolution in interleaved mode. I'm pretty sure it will break the base map rendering. What is the use case?

File: modules/mapbox/src/deck-utils.ts [hunk 1]
```diff
@@ -112,11 +112,8 @@ export function removeDeckInstance(map: Map & {__deck?: Deck | null}) {
 }
 
 export function getInterleavedProps(currProps: DeckProps) {
-  const useDevicePixels = 'useDevicePixels' in currProps ? currProps.useDevicePixels : true;
-
   const nextProps: DeckProps = {
     ...currProps,
-    useDevicePixels,
```
Comment (chrisgervang) [inline modules/mapbox/src/deck-utils.ts:119]: Resolution can currently be controlled in two parts. One, by setting the width/height css style of the react-map-gl HTML container, and two, multiplying width/height by setting `useDevicePixels` prop on deck and `pixelRatio` prop on react-map-gl for maplibre. 

Was this redundant or does this change modify behavior?

Comment (Pessimistress) [reply modules/mapbox/src/deck-utils.ts:119]: You should be able to just set `pixelRatio` on the base map after this PR.

**Review by chrisgervang — APPROVED** [review]

**Pull request merged**